bridge/launchpad/import.go

```diff
@@ -45,98 +44,116 @@ func (li *launchpadImporter) ensurePerson(repo *cache.RepoCache, owner LPPerson)
 	)
 }
 
-func (li *launchpadImporter) ImportAll(repo *cache.RepoCache, since time.Time) error {
+func (li *launchpadImporter) ImportAll(ctx context.Context, repo *cache.RepoCache, since time.Time) (<-chan core.ImportResult, error) {
+	out := make(chan core.ImportResult)
 	lpAPI := new(launchpadAPI)
 
 	err := lpAPI.Init()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	lpBugs, err := lpAPI.SearchTasks(li.conf["project"])
+	lpBugs, err := lpAPI.SearchTasks(ctx, li.conf["project"])
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	for _, lpBug := range lpBugs {
-		var b *cache.BugCache
-		var err error
-
-		lpBugID := fmt.Sprintf("%d", lpBug.ID)
-		b, err = repo.ResolveBugCreateMetadata(keyLaunchpadID, lpBugID)
-		if err != nil && err != bug.ErrBugNotExist {
-			return err
-		}
-
-		owner, err := li.ensurePerson(repo, lpBug.Owner)
-		if err != nil {
-			return err
-		}
-
-		if err == bug.ErrBugNotExist {
-			createdAt, _ := time.Parse(time.RFC3339, lpBug.CreatedAt)
-			b, _, err = repo.NewBugRaw(
-				owner,
-				createdAt.Unix(),
-				lpBug.Title,
-				lpBug.Description,
-				nil,
-				map[string]string{
-					keyLaunchpadID: lpBugID,
-				},
-			)
-			if err != nil {
-				return errors.Wrapf(err, "failed to add bug id #%s", lpBugID)
+	go func() {
+		for _, lpBug := range lpBugs {
+			select {
+			case <-ctx.Done():
```

Comment:
same, do you really need that select ?

Reply:
`lpAPI.SearchTasks` query multiple tasks a time, we need to check context cancellation at each iteration